src/Features/CSharp/Portable/CodeRefactorings/ConvertLocalFunctionToMethod/CSharpConvertLocalFunctionToMethodCodeRefactoringProvider.cs

```diff
@@ -42,41 +43,54 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
             var cancellationToken = context.CancellationToken;
             var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
 
-            LocalFunctionStatementSyntax localFunction = default;
-            if (context.Span.IsEmpty)
+            var localFunction = await TryGetSelectedLocalFunction(context.Span, root, cancellationToken).ConfigureAwait(false);
+            if (localFunction == default)
+            {
+                return;
+            }
+
+            if (!localFunction.Parent.IsKind(SyntaxKind.Block, out BlockSyntax parentBlock))
+            {
+                return;
+            }
+
+            context.RegisterRefactoring(new MyCodeAction(CSharpFeaturesResources.Convert_to_method,
+                c => UpdateDocumentAsync(root, document, parentBlock, localFunction, c)));
+        }
+
+        private async Task<LocalFunctionStatementSyntax> TryGetSelectedLocalFunction(TextSpan selectedSpan, SyntaxNode root, CancellationToken cancellationToken)
+        {
+            if (selectedSpan.IsEmpty)
             {
                 // This branch is required to handle "C LocalFunction[||](C c)" -> root.FindNode return ParameterList
-                var identifier = await root.SyntaxTree.GetTouchingTokenAsync(context.Span.Start, token => token.Parent.IsKind(SyntaxKind.LocalFunctionStatement), cancellationToken).ConfigureAwait(false);
-                if (identifier == default)
-                {
-                    return;
-                }
-                localFunction = (LocalFunctionStatementSyntax)identifier.Parent;
+                var identifier = await root.SyntaxTree.GetTouchingTokenAsync(selectedSpan.Start,
+                    token => token.Parent.IsKind(SyntaxKind.LocalFunctionStatement), cancellationToken).ConfigureAwait(false);
+
+                return (identifier == default)
+                    ? null
+                    : (LocalFunctionStatementSyntax)identifier.Parent;
             }
             else
```

Comment:
no need for else

Reply:
The idea behind having it explicit like this is communicating the intent of the code better while doing only little (unnecessary) work. Happy to fix it, however.